openstack/compute/v2/extensions/services/requests.go

```diff
@@ -11,3 +11,34 @@ func List(client *gophercloud.ServiceClient) pagination.Pager {
 		return ServicePage{pagination.SinglePageBase(r)}
 	})
 }
+
+// UpdateOpts specifies the base attributes that may be updated on a service.
+type UpdateOpts struct {
+	// Status represents the new service status. One of enabled or disabled.
+	Status string `json:"status,omitempty"`
```

Comment:
Since this is input (part of a request) and there are finite choices, we can use a defined type:

```go
type ServiceStatus string

const (
  // ServiceEnabled is used to mark a service as being enabled.
  ServiceEnabled ServiceStatus = "enabled"

  // ServiceDisabled is used to mark a service as being disabled.
  ServiceDisabled ServiceStatus = "disabled"
)

type UpdateOpts struct {
  ...
  Status ServiceStatus `json:"status,omitempty"`
  ...
}
```

The `Status` field in the `results.go` will stay a string, though.

Reply:
Yea, that looks better.  I'll make that change.